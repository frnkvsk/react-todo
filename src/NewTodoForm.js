/**
 NewTodoForm - this component should render a form with one text input for the task to be created. When this form is submitted, a new Todo component should be created.
 */
import React, {useState} from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'grid',
    gridTemplateColumns: '4fr 1fr',
    gridTemplateRows: 'auto',
    gridTemplateAreas:
      'TextField Button',
    width: '100%'

  },
  button: {
    background: 'blue',
    color: 'white',
    marginLeft: '5px'
  }
}));

const NewTodoForm = ({addNewTodo}) => {
  const initialState = {
    todo: "",
  }
  const [formData, setFormData] = useState(initialState)
  const handleChange = e => {
    const {name, value} = e.target;
    setFormData(data => ({
      ...data,
      [name]: value
    }));
  }
  const handleSubmit = e => {
    e.preventDefault();
    if(formData.todo.length) {
      addNewTodo(formData.todo);
      setFormData(initialState);
    }    
  }
  const classes = useStyles();

  return (
    <form onSubmit={handleSubmit} className={classes.container} noValidate autoComplete="off">
      <TextField 
        id="form-text-color" 
        label="Add a new todo"
        variant="outlined" 
        name="todo"
        value={formData.todo}
        onChange={handleChange} />
      <Button type="submit" id="form-btn" variant="contained" color="primary" className={classes.button}>Add Todo</Button>
    </form>
  );
}

export default NewTodoForm;