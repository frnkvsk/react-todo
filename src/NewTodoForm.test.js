import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

describe("smoke and snapshot", () => {
  const mockAddNewTodo = jest.fn();
  // smoke test
  it("Control renders without crashing", () => {
    render(<NewTodoForm addNewTodo={mockAddNewTodo} />);
  });

  // snapshot test
  it("matches snapshot", () => {
    const {asFragment} = render(<NewTodoForm addNewTodo={mockAddNewTodo} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("check for initial components", () => {
  const mockAddNewTodo = jest.fn();
  let { getByLabelText } = render(<NewTodoForm addNewTodo={mockAddNewTodo} />);
  
  const textInput = screen.getByLabelText('Add a new todo');
  expect(textInput).toBeInTheDocument();
  
  const button = screen.queryByText(/ADD TODO/i);
  expect(button).toBeInTheDocument();
});

describe("check input field", () => {
  it("check for input", async () => {
    const mockAddNewTodo = jest.fn({todo:"do testing"});
    const handleSubmit = jest.fn();

    let { getByLabelText } = render(<NewTodoForm addNewTodo={mockAddNewTodo} />);
  
    const textInput = screen.getByLabelText('Add a new todo');
    fireEvent.change(textInput, { target: { value: "do test"}});
    expect(textInput).toHaveAttribute('value', 'do test');


    const button = screen.queryByText(/ADD TODO/i);
    await fireEvent.click(button);
    // can't figure out how to test handleSubmit or addNewTodo function calls from form submit

  });
});
