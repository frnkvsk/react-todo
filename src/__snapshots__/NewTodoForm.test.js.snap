// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot matches snapshot 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="makeStyles-container-7"
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
        for="form-text-color"
        id="form-text-color-label"
      >
        Add a new todo
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="form-text-color"
          name="todo"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-11"
          >
            <span>
              Add a new todo
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary"
      id="form-btn"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Add Todo
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
